docs/02-app/01-building-your-application/06-optimizing/10-third-party-libraries.mdx

```diff
@@ -0,0 +1,288 @@
+> active development. We're currently working on adding more third-party integrations.
+
+All third-party integrations provided by `@next/third-parties` have been optimized for performance
+and ease of use.
+
+## Getting Started
+
+To get started, you must install the `@next/third-parties` library:
+
+```bash filename="Terminal"
+npm install @next/third-parties
+```
+
+## Google Third-Parties
+
+All supported third-party libraries from Google can be imported from `@next/third-parties/google`.
+
+### Google Tag Manager
+
+The `GoogleTagManager` component can be used to instantiate a [Google Tag
+Manager](https://developers.google.com/maps/documentation/embed/embedding-map) container to your
+page. By default, it fetches the original inline script after hydration occurs on the page.
+
+<AppOnly>
+
+To load Google Tag Manager for all routes, include the component directly in your root layout:
+
+```tsx filename="app/layout.tsx" switcher
+import { GoogleTagManager } from '@next/third-parties/google'
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  return (
+    <html lang="en">
+      <body>{children}</body>
+      <GoogleTagManager gtmId="GTM-XYZ" />
+    </html>
+  )
+}
+```
+
+```jsx filename="app/layout.js" switcher
+import { GoogleTagManager } from '@next/third-parties/google'
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <body>{children}</body>
+      <GoogleTagManager gtmId="GTM-XYZ" />
+    </html>
+  )
+}
+```
+
+</AppOnly>
+
+<PagesOnly>
+
+To load Google Tag Manager for all routes, include the component directly in your custom `_app`:
+
+```jsx filename="pages/_app.js"
+import { GoogleTagManager } from '@next/third-parties/google'
+
+export default function MyApp({ Component, pageProps }) {
+  return (
+    <>
+      <Component {...pageProps} />
+      <GoogleTagManager gtmId="GTM-XYZ" />
+    </>
+  )
+}
+```
+
+</PagesOnly>
+
+To load Google Tag Manager for a single route, include the comopnent in your page file:
+
+<AppOnly>
+
+```jsx filename="app/page.js"
+import { GoogleTagManager } from '@next/third-parties/google'
+
+export default function Page() {
+  return <GoogleTagManager gtmId="GTM-XYZ" />
+}
+```
+
+</AppOnly>
+
+<PagesOnly>
+
+```jsx filename="pages/index.js"
+import { GoogleTagManager } from '@next/third-parties/google'
+
+export default function Page() {
+  return <GoogleTagManager gtmId="GTM-XYZ" />
+}
+```
+
+</PagesOnly>
+
+#### Sending Events
+
+The `sendGTMEvent` function can be used to track user interactions on your page by sending events
+using the `dataLayer` object. For this function to work, the `<GoogleTagManager />` component must be
+included in either a parent layout, page, or component, or directly in the same file.
+
+<AppOnly>
+
+```jsx filename="app/page.js"
+'use client'
+
+import { sendGTMEvent } from '@next/third-parties/google'
```

Comment:
```suggestion
import { sendGtmEvent } from '@next/third-parties/google'
```

Can this function be named using `strictCamelCase`? Or is too late? Usage of `strictCamelCase` can be enforced in some codebase [via ESLint](https://typescript-eslint.io/rules/naming-convention/#format-options).

Reply:
If it’s too late before the initial release, it’s still possible to make the shift to `strictCamelCase` later. It is possible to mark `export const sendGTMEvent` as deprecated and expose `export const sendGtmEvent` alongside it. Docs can only mention `sendGtmEvent`.

Reply:
Yeah I think changing the name to `sendGtmEvent` makes total sense. We'll likely do that after the initial release if that's alright (and since the package is still marked experimental it shouldn't be too much of a problem). @janicklas-ralph wdyt?